dask/delayed.py

```diff
@@ -472,9 +471,12 @@ def _inner(self, other):
 
 
 def optimize(dsk, keys, **kwargs):
-    dsk = ensure_dict(dsk)
-    dsk2, _ = cull(dsk, keys)
-    return dsk2
+    if not isinstance(keys, (list, set)):
+        keys = [keys]
+    if not isinstance(dsk, HighLevelGraph):
+        dsk = HighLevelGraph.from_collections(id(dsk), dsk, dependencies=())
```

Comment:
I was surprised to see we still need to support low-level task graphs here. Can you remind me why this is needed?

Reply:
I'm not sure that it is need anymore, to be honest. This stanza is from 2a51476, which looks like it was copied from `array` and `dataframe`, both of which have similar defensiveness.

Reply:
Turns out there are still some places where delayed uses low-level graphs -- quite a number of tests fail without this check.

I'll bet it could be fixed by moving the HighLevelGraph cast to those places. Do you have any idea of how much bespoke user code there might be out there manually constructing Delayed graphs?

Reply:
Ah, I see. My guess is there's probably not much manual constructing of `Delayed` graphs

Reply:
Yeah it might be interesting to put a WIP PR and see how many tests fail. At a minimum it would provide a good list of things to add HLGs to

Reply:
I took a look earlier today, and a good many of the failures seemed to come from [this](https://github.com/dask/dask/blob/6353103d68e9d98fc103a0967744763f6d130eb6/dask/base.py#L780) manual graph in `persist`.

I don't think it would be a huge lift (for delayed, at least).

Reply:
Just so I understand, we want to move that to HLG? If so, agreed that looks doable and pretty valuable actually